sdk/src/main/java/com/stormpath/sdk/ApiManager.java

```diff
@@ -0,0 +1,360 @@
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.logging.HttpLoggingInterceptor;
+
+public class ApiManager {
+
+    public static final Pattern ACCESS_TOKEN_COOKIE_PATTERN = Pattern.compile("access_token=(.*?);.*");
+
+    public static final Pattern REFRESH_TOKEN_COOKIE_PATTERN = Pattern.compile("refresh_token=(.*?);.*");
+
+    private OkHttpClient okHttpClient;
+
+    private Executor callbackExecutor;
+
+    private StormpathConfiguration config;
+
+    private PreferenceStore preferenceStore;
+
+    private Moshi moshi = new Moshi.Builder().build();
+
+    ApiManager(StormpathConfiguration config, Platform platform) {
+        this.config = config;
+        this.preferenceStore = platform.preferenceStore();
+        this.callbackExecutor = platform.callbackExecutor();
+        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor(new HttpLoggingInterceptor.Logger() {
+            @Override
+            public void log(String message) {
+                Stormpath.logger().d(message);
+            }
+        });
+        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+        this.okHttpClient = new OkHttpClient.Builder()
+                .dispatcher(new Dispatcher(platform.httpExecutorService()))
+                .addNetworkInterceptor(httpLoggingInterceptor)
+                .build();
+    }
+
+    void login(String username, String password, StormpathCallback<Void> callback) {
+        RequestBody formBody = new FormBody.Builder()
+                .add("username", username)
+                .add("password", password)
+                .add("grant_type", "password")
+                .build();
+
+        Request request = new Request.Builder()
+                .url(config.oauthUrl())
+                .header("Accept", "application/json")
+                .post(formBody)
+                .build();
+
+        okHttpClient.newCall(request).enqueue(new OkHttpCallback<Void>(callback) {
+            @Override
+            protected void onSuccess(Response response, StormpathCallback<Void> callback) {
+                try {
+                    LoginResponse loginResponse = moshi.adapter(LoginResponse.class).fromJson(response.body().source());
+                    if (StringUtils.isBlank(loginResponse.getAccessToken())) {
+                        failureCallback(new RuntimeException("access_token was not found in response. See debug logs for details."));
+                        return;
+                    }
+
+                    if (StringUtils.isBlank(loginResponse.getRefreshToken())) {
+                        Stormpath.logger().e("There was no refresh_token in the login response!");
+                    }
+
+                    preferenceStore.setAccessToken(loginResponse.getAccessToken());
+                    preferenceStore.setRefreshToken(loginResponse.getRefreshToken());
+                    successCallback(null);
```

Comment:
might be my ignorance here - where is `successCallback` and `failureCallback` coming from?